algorithms/README.md

```diff
@@ -1,222 +1,212 @@
-4. **Задача о покрытии множества.** Например, необходимо выбрать минимальное количество радиостанций, которые будут покрывать максимально большое количество областей. **Решение:** Выбрать станцию с наибольшим покрытием. Если станция будет покрывать области, которые уже были покрыты, то это нормально. Повторять, пока остаются штаты, не входящие в покрытие.
-5. **Задача о коммивояжере.** Коммивояжер должен был посетить пять разных городов. Коммивояжер должен найти кратчайший путь, который включит все города. **Решение:** Проходим по всем возможным комбинациям путей и выбираем минимальное расстояние.
+Problems suitable for greedy algorithms:
+1. **Scheduling Problem**: For instance, you have a class schedule with overlapping lessons. It's impossible to conduct all lessons because they clash in time. **Solution:** Choose the lesson that ends first, then find the next lesson that starts immediately after the previous one, and so on.
+2. **Coin Change Problem**: You have a set of coins with different denominations, and you need to make change for a given amount with the fewest number of coins. **Solution:** Use as many coins with the highest denomination as possible, then move to the lower denomination coins, and so on.
+3. **Knapsack Problem**: A thief with a limited capacity knapsack wants to steal the most valuable items from a store. **Solution:** Choose the most valuable item and place it in the knapsack. Continue until the knapsack is full.
+4. **Set Cover Problem**: Suppose you need to select the minimum number of radio stations to cover the maximum number of areas. **Solution:** Choose the station with the most extensive coverage. If it covers some areas already covered, it's fine. Repeat until all areas are covered.
+5. **Traveling Salesman Problem**: A traveling salesman needs to visit five different cities. The salesman must find the shortest path that includes all cities. **Solution:** Iterate through all possible paths and select the one with the shortest distance.
 
-_Важно!_ Далеко не всегда жадный алгоритм эффективный и его необходимо применять. Например в задаче о рюкзаке с воришкой могла быть ситуация, что первый же товар займет все место в рюкзаке. Однако, было бы сильно выгоднее, набрать товара меньшего размера, но в больше количестве. Если это имело место быть, то алгоритм требовал бы учитывать не только стоимость товара, но и размер.
+_Important!_ Greedy algorithms are not always efficient, and their application should be considered carefully. For example, in the knapsack problem, it's possible that the first item selected occupies the entire knapsack's capacity. However, it might be more profitable to choose smaller-sized items in greater quantities. In such cases, the algorithm would need to consider not only the item's value but also its size.
```

Comment:
However, it might be more profitable to choose smaller-sized items in greater quantities -> And it might be more profitable to choose smaller-sized items in greater quantities.
 
 Тут я бы и на русском "однако" заменила